lib/plugin-loader.js

```diff
@@ -752,71 +756,275 @@ PluginLoader.prototype = {
+            resolve(); // We don't want to reject here. Error gets caught down stream
+          }
+        });
+      }).on('error', (e) => {
+        bootstrapLogger.severe(e.message);
+        resolve();
+      });
+    }).then(() => { /* Obtains and stores the endpoints exposed by the agent */
+
+      'http://' + host + ':' + port + '/server/agent/services';
+
+      return new Promise((resolve, reject) => { 
+        http.get(url, (res) => {
+        const { statusCode } = res; // TODO: Check status code for bad status
+        const contentType = res.headers['content-type'];
+
+        res.setEncoding('utf8');
+        let rawData = '';
+        res.on('data', (chunk) => { rawData += chunk; });
+        res.on('end', () => {
+          try {
+            const parsedData = JSON.parse(rawData);
+            if (parsedData.services) {
+              agentComp.endpoints = [];
+              for (let i = 0; i < parsedData.services.length; i++) {
+                if (parsedData.services[i].urlMask) {
+                  agentComp.endpoints.push(parsedData.services[i].urlMask);
+                }
+              }
+            }
+            resolve();
+          } catch (e) {
+            bootstrapLogger.severe(e.message);
+            resolve(); // We don't want to reject here. Error gets caught down stream
+          }
+        });
+        }).on('error', (e) => {
+          bootstrapLogger.severe(e.message);
+          resolve(); // We don't want to reject here. Error gets caught down stream
+        });
+      }).then(() => {
+        let obj = {};
+        obj[AGENT_COMP_ID] = agentComp;
+        obj[APP_SERVER_COMP_ID] = appServerComp;
+        return obj;
+      });
     });
-    for (const rejectedPlugin of sortedAndRejectedPlugins.rejects) {
-      const rejectionError = zluxUtil.formatErrorStatus(rejectedPlugin.validationError, DependencyGraph.statuses);
-      bootstrapLogger.warn(`ZWED0033W`, rejectedPlugin.pluginId, rejectionError); //bootstrapLogger.warn(`Could not initialize plugin` 
-          //+ ` ${rejectedPlugin.pluginId}: `  
-          //+ zluxUtil.formatErrorStatus(rejectedPlugin.validationError, 
-              //DependencyGraph.statuses));
-      newPlugins.push(Object.assign(rejectedPlugin, {error: rejectionError}));
-    }
+  }
+},
 
-    let isFirstRun = Object.keys(this.pluginMap).length === 0;
-    
-    for (const pluginDef of sortedAndRejectedPlugins.plugins) { 
-      try {
-        if (this.pluginMap[pluginDef.identifier]) {
-          bootstrapLogger.warn(`ZWED0034W`, plugin.identifier); //bootstrapLogger.warn(`Skipping install of plugin due to existing plugin with same id=${plugin.identifier}`);
+  checkPluginRequirements(plugin, components) {
+    if (plugin.pluginReqs) {
+      const acceptedComponents = plugin.pluginReqs.components;
+      
+      if (acceptedComponents) {
+        if (acceptedComponents[APP_SERVER_COMP_ID]) { /* --- App server component --- */
```

Comment:
honestly this `checkPluginRequirements` function has no need for explicit component ID checking. Any such need happened in the gathering of component info stage, this can just run through some `Object.keys` instead.